### PR TITLE
[README] Clarify note about unneeded dependencies of Poetry

### PR DESCRIPTION
**Pull request by homeworkprod**
`poetry-core` [is a dependency of Poetry](https://github.com/python-poetry/poetry/blob/eb80d10846f7336b0b2a66ce2964e72dffee9a1c/pyproject.toml#L35), of course.

This change should avoid confusion arising from the former wording.

### CODE REVIEW

**Review by radoering — APPROVED** [review]

**Comment by radoering** [comment]
LGTM. However, it seems you did not check "Allow edits from maintainers". Thus, I cannot rebase your branch and finally cannot merge it. Please rebase your branch yourself or set the checkbox.

**Comment by homeworkprod** [comment]
Rebased.

> However, it seems you did not check "Allow edits from maintainers".

Sorry about that. I've heard this remark before in another project, but in both cases that checkbox was/is nowhere to be seen; at least not on the PR page itself. Not sure if this is somehow dependent on how actually a PR is created (e.g. directly on a repository vs. off of another repository).

**Comment by homeworkprod** [comment]
According to https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork the reason this doesn't work here is that my PR comes from a non-user-owned fork (I put forks in a separate org so they don't clutter up my "actual" projects, a practice I've copied from somewhere).

**Comment by homeworkprod** [comment]
Also see here for an unresolved discussion: https://github.com/orgs/community/discussions/5634

**Pull request merged**
